src/action/SummationAction.cpp

```diff
@@ -0,0 +1,44 @@
+#include "SummationAction.hpp"
+#include "mesh/Data.hpp"
+#include "mesh/Edge.hpp"
+#include "mesh/Mesh.hpp"
+
+namespace precice {
+namespace action{
+
+SummationAction::SummationAction(
+	Timing				timing,
+	std::vector<int>	sourceDataID,
+	int					targetDataID,
+	const mesh::PtrMesh	&mesh)
+	: Action(timing, mesh, mapping::Mapping::MeshRequirement::VERTEX), _targetData(mesh->data(targetDataID))
+{
+
+	for(auto sourceID : sourceDataID){
```

Comment:
```suggestion
	for(int sourceDataID : sourceDataIDs){
```
I would recommend to avoid `auto` if it is just a simple type, such as `int`. Probably my personal taste, confuses me more than it helps.